src/app/pools/components/PoolForm/PoolForm.test.tsx

```diff
@@ -62,36 +60,31 @@ describe("PoolForm", () => {
       </Provider>
     );
 
-    act(() => {
-      wrapper.unmount();
-    });
+    unmount();
 
     expect(store.getActions()[0]).toEqual(actions.cleanup());
   });
 
   it("redirects when the resource pool is saved", () => {
     state.resourcepool.saved = true;
-    const store = mockStore(state);
-
-    const wrapper = mount(
-      <Provider store={store}>
-        <MemoryRouter initialEntries={["/pool/add"]}>
-          <CompatRouter>
-            <PoolForm />
-          </CompatRouter>
-        </MemoryRouter>
-      </Provider>
-    );
-    expect(wrapper.find(Router).prop("history").location.pathname).toBe(
-      urls.pools.index
+    const history = createMemoryHistory({
+      initialEntries: ["/"],
+    });
+    renderWithMockStore(
+      <Router history={history}>
+        <CompatRouter>
+          <PoolForm />
+        </CompatRouter>
+      </Router>,
+      { state }
     );
+    expect(history.location.pathname).toBe(urls.pools.index);
```

Comment:
If you wanted to use `renderWithBrowserRouter` for this test you'd need to use the window history, like in this test: https://github.com/canonical/maas-ui/blob/main/src/app/settings/components/Routes/Routes.test.tsx#L214